docker/nginx/nginx-key.conf

```diff
@@ -64,155 +53,62 @@ http {
-            proxy_set_header X-Forwarded-Proto $scheme;
+            include snippets/proxy-headers.conf;
             add_header Access-Control-Expose-Headers Location;
-            proxy_pass http://things:8182;
-
-            # Allow OPTIONS method CORS
-            if ($request_method = OPTIONS ) {
-                add_header Content-Length 0;
-                add_header Content-Type text/plain;
-                return 200;
-            }
+            proxy_pass http://things:${MF_THINGS_HTTP_PORT};
         }
 
         location /version {
-            proxy_redirect off;
-            proxy_set_header Host $host;
-            proxy_set_header X-Real-IP $remote_addr;
-            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
-            proxy_set_header X-Forwarded-Proto $scheme;
-            proxy_pass http://things:8182;
-
-            # Allow OPTIONS method CORS
-            if ($request_method = OPTIONS ) {
-                add_header Content-Length 0;
-                add_header Content-Type text/plain;
-                return 200;
-            }
+            include snippets/proxy-headers.conf;
+            proxy_pass http://things:${MF_THINGS_HTTP_PORT};
         }
 
         # Proxy pass to mainflux-http-adapter
         location /http/ {
-            proxy_redirect off;
-            proxy_set_header Host $host;
-            proxy_set_header X-Real-IP $remote_addr;
-            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
-            proxy_set_header X-Forwarded-Proto $scheme;
-            proxy_pass http://http-adapter:8185/;
-
-            # Allow OPTIONS method CORS
-            if ($request_method = OPTIONS ) {
-                add_header Content-Length 0;
-                add_header Content-Type text/plain;
-                return 200;
-            }
+            include snippets/proxy-headers.conf;
+            proxy_pass http://http-adapter:${MF_HTTP_ADAPTER_PORT};
         }
 
         # Proxy pass to mainflux-ws-adapter
         location /ws/ {
-            proxy_redirect off;
-            proxy_set_header Host $host;
-            proxy_set_header X-Real-IP $remote_addr;
-            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
-            proxy_set_header X-Forwarded-Proto $scheme;
-            proxy_http_version 1.1;
-            proxy_set_header Upgrade $http_upgrade;
-            proxy_set_header Connection "Upgrade";
-            proxy_connect_timeout 7d;
-            proxy_send_timeout 7d;
-            proxy_read_timeout 7d;
-            proxy_pass http://ws-adapter:8186/;
-
-            # Allow OPTIONS method CORS
-            if ($request_method = OPTIONS ) {
-                add_header Content-Length 0;
-                add_header Content-Type text/plain;
-                return 200;
-            }
+            include snippets/proxy-headers.conf;
+            include snippets/ws-upgrade.conf;
+            proxy_pass http://ws-adapter:${MF_WS_ADAPTER_PORT};
```

Comment:
I think you missed '/' here

Reply:
Thanks, good catch. Added this with a comment.